### PR TITLE
Refactor and handle new network qubesdb configuration

### PR DESCRIPTION
**Pull request by fepitre**
WIP: for testing with OpenQA

### CODE REVIEW

File: network/setup-ip [hunk 1]
```diff
@@ -1,135 +1,231 @@
...
+#!/bin/bash
 
 # Source Qubes library.
 # shellcheck disable=SC1091
 . /usr/lib/qubes/init/functions
 
-have_qubesdb || exit 0
+configure_network() {
+    local IFS=','
+    read -r -a args <<< "$@"
```
[the remark this answers is not in the document]
Comment (fepitre) [reply network/setup-ip]: it's for handling empty parameters

[the remark this answers is not in the document]
Comment (marmarek) [reply network/setup-ip]: Quoting parameters (as you already do, in addition to colons) is enough for this too.

**Comment by marmarek** [comment]
OpenQA run is here: https://openqa.qubes-os.org/tests/overview?distri=qubesos&version=4.0&build=4.0-20190619.2&groupid=6
It is R4.0 + all updates from current-testing + this commit on top of release4.0 branch.

It looks like network doesn't at all, see for example https://openqa.qubes-os.org/tests/2551
In logs tab, you'll see var_log.tar.gz - see `var/log/xen/console/guest-sys-net.log` there. Journalctl is redirected to console, so you'll have all the VMs logs there. I've found this:
```
[2019-06-19 07:12:01]          Starting Qubes network forwarding setup...
[2019-06-19 07:12:01] [   17.379794] network-proxy-setup.sh[452]: Failed to read /qubes-netvm-gateway6
[2019-06-19 07:12:01] [   17.579473] audit: type=1325 audit(1560942721.123:30): table=nat family=2 entries=15
[2019-06-19 07:12:01] [   17.739313] network-proxy-setup.sh[452]: Failed to read /qubes-mac
[2019-06-19 07:12:01] [   17.840044] network-proxy-setup.sh[452]: Cannot get device feature names: No such device
[2019-06-19 07:12:01] [   17.843192] systemd[1]: Started Qubes network forwarding setup.
```
But maybe there is more.

**Comment by marmarek** [comment]
Next test run (with https://github.com/QubesOS/qubes-core-agent-linux/pull/163/commits/1c5410166320ff55a25b8675b8ee2c6342d6d9ba):
sys-net:
```
[2019-06-19 16:19:59] [   18.028920] network-proxy-setup.sh[482]: Failed to read /qubes-netvm-gateway6
[2019-06-19 16:19:59] [   18.188349] audit: type=1325 audit(1560975599.811:34): table=nat family=2 entries=15
[2019-06-19 16:20:00] [   18.319304] network-proxy-setup.sh[482]: Failed to read /qubes-mac
[2019-06-19 16:20:00] [   18.354076] network-proxy-setup.sh[482]: Cannot get device feature names: No such device
...
[2019-06-19 16:20:00] [   18.798507] misc-post.sh[519]: Failed to read /qubes-mac
[2019-06-19 16:20:00] [   18.910384] misc-post.sh[519]: cat: /sys/class/net/eth0/address: No such file or directory
```
This is sys-net, which isn't connected to another network-providing VM. In fact eth0 doesn't exist there. The above may be harmless, but IMO there should be some clean check for this earlier.

File: network/setup-ip [hunk 2]
```diff
@@ -1,135 +1,225 @@
...
+            # Handle legacy values
+            LEGACY_MAC="$(/usr/bin/qubesdb-read /qubes-mac 2> /dev/null)"
+            if [ "$MAC" == "$LEGACY_MAC" ] || [ "x$LEGACY_MAC" == "x" ]; then
+                if [ "x$ip" == "x" ]; then
+                    ip="$(/usr/bin/qubesdb-read /qubes-ip 2> /dev/null)"
+                fi
+                if [ "$ip6" == "x" ]; then
+                    ip6="$(/usr/bin/qubesdb-read /qubes-ip6 2> /dev/null)"
+                fi
+                if [ "$gateway" == "x" ]; then
```
Comment (marmarek) [inline network/setup-ip]: This lack "x" on the left side. Result: no default gateway set.
Shouldn't this be `[ -z "$gateway" ]` instead? (and the same for others)

**Pull request merged**